template.tpl

```diff
@@ -1171,7 +1231,8 @@ let platform = eventData['x-sp-platform'] || data.fallbackPlatform;
 
 let amplitudeEvent = {
   event_type: eventData.event_name,
-  device_id: eventData.client_id,
+  device_id: data.defaultDeviceId ? eventData.client_id : data.deviceId,
+  user_id: data.defaultUserId ? eventData.user_id : data.userId,
```

Comment:
This is a slight behaviour change I think. We used to ensure `eventData.user_id` wasn't falsey before setting it previously (i.e. an empty string).

Reply:
(This change might be fine, but I figured I included the `if` check previously because Amplitude prefers an undefined property over an empty string.)

Reply:
Thank you for reviewing @paulboocock !
This was my fault, indeed it was a change in behaviour. I changed it to set it to `undefined`, if `event_data.user_id` is falsy (which will then get removed when stringified).